src/mpid/ch4/shm/posix/eager/fbox/fbox_types.h

```diff
@@ -19,12 +19,12 @@
 
 typedef struct {
 
-    volatile uint64_t data_ready;
+    volatile uint64_t data_ready;       /*  takes a cache line */
 
     int is_header;
-    size_t payload_sz;
+    unsigned int payload_sz;    /* payload and cell can't be more than 4GB */
 
```

Comment:
"can't be more than 4GB" because of `unsigned int` or because it is the semantics of "fast box"? If latter, I would imagine 4GB is still way too big?

Reply:
Again, it is just a comment because I downsized payload_sz from size_t to int. 
4GB is the limit, but like you said, it should not be an issue.